### PR TITLE
Refactoring and updating convertIfcHalfSpaceSolid function

### LINKED ISSUE #438: [REF] convertIfcHalfSpaceSolid
**File which is being refactored**
core/IfcGeometryConverter/SolidConverter

**Functions**
convertIfcHalfSpaceSolid

### PR DESCRIPTION
**Pull request by SamuilsRulovs**
fixes #438

### CODE REVIEW

**Review by jschlenger — COMMENTED** [review]
Good start! On first sight everything seems to be fine.

File: Core/src/IfcGeometryConverter/SolidModelConverter.h [hunk 1]
```diff
@@ -2259,126 +2258,119 @@ namespace OpenInfraPlatform
...
+						extrudeBox(baseSurfacePoints, halfSpaceExtrusionVector, halfSpaceBoxData);
 
 						// apply object coordinate system
-						for (std::vector<carve::geom::vector<3> >::iterator it_points = half_space_box_data->points.begin(); it_points != half_space_box_data->points.end(); ++it_points)
+						for ( auto point : halfSpaceBoxData->points)
 						{
-							carve::geom::vector<3> & poly_point = (*it_points);
-							poly_point = pos * poly_point;
+							carve::geom::vector<3> & polyPoint = pos * point;
 						}
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/SolidModelConverter.h]: This changed the previous meaning.

It should say

for( auto& point : halfSpaceBoxData->points )
point = pos * point;

File: Core/src/IfcGeometryConverter/SolidModelConverter.h [hunk 2]
```diff
@@ -2259,126 +2258,119 @@ namespace OpenInfraPlatform
...
+						extrudeBox(boxBasePoints, halfSpaceExtrusionVector, halfSpaceBoxData);
 
 						// apply object coordinate system
-						for (std::vector<carve::geom::vector<3> >::iterator it_points = half_space_box_data->points.begin(); it_points != half_space_box_data->points.end(); ++it_points)
+						for (auto point : halfSpaceBoxData->points)
 						{
-							carve::geom::vector<3> & poly_point = (*it_points);
-							poly_point = pos * poly_point;
+							carve::geom::vector<3> & poly_point = pos * point;
 						}
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/SolidModelConverter.h]: Similarly.

File: Core/src/IfcGeometryConverter/SolidModelConverter.h [hunk 3]
```diff
@@ -2416,105 +2408,102 @@ namespace OpenInfraPlatform
 
 				itemData->closed_polyhedrons.push_back(polyhedron_data);
 				// apply object coordinate system
-				for (std::vector<carve::geom::vector<3>>::iterator it_points = polyhedron_data->points.begin(); it_points != polyhedron_data->points.end(); ++it_points)
+				for ( auto point : polyhedron_data->points)
 				{
-					carve::geom::vector<3> & poly_point = (*it_points);
-					poly_point = box_position_matrix * poly_point;
+					 carve::geom::vector<3> poly_point = box_position_matrix * point;
 				}
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/SolidModelConverter.h]: Similarly.

File: Core/src/IfcGeometryConverter/SolidModelConverter.h [hunk 4]
```diff
@@ -2524,25 +2513,24 @@ namespace OpenInfraPlatform
...
 
 				// apply object coordinate system
-				for (std::vector<carve::geom::vector<3> >::iterator it_points = poly_data->points.begin(); it_points != poly_data->points.end(); ++it_points)
-				{
-					carve::geom::vector<3>& poly_point = (*it_points);
-					poly_point = pos * poly_point;
+				for (auto point : polyData->points)
+					{
+					carve::geom::vector<3>& poly_point = pos * point;
 				}
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/SolidModelConverter.h]: Yet again.

**Review by pjanck — CHANGES_REQUESTED** [review]
I'm unsure if this works. Have you tried any example files?

File: Core/src/IfcGeometryConverter/SolidModelConverter.h (hunk 4 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/SolidModelConverter.h]: Resolved: 01cd5dc

File: Core/src/IfcGeometryConverter/SolidModelConverter.h (hunk 2 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/SolidModelConverter.h]: Resolved: 01cd5dc

File: Core/src/IfcGeometryConverter/SolidModelConverter.h (hunk 3 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/SolidModelConverter.h]: Resolved: 01cd5dc

File: Core/src/IfcGeometryConverter/SolidModelConverter.h (hunk 1 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/SolidModelConverter.h]: Resolved: 01cd5dc

**Review by jschlenger — APPROVED** [review]
Looks good!

**Review by pjanck — APPROVED** [review]
Does it work, though? If yes, this can be merged.

**Comment by SamuilsRulovs** [comment]
We don't have any example files to check, if it works. I can make an Issue about that

**Pull request merged**